### ISSUE TITLE
Please backport 287ce6b51675aee43870bebfff68bb144d1ab90e ([lldb] [Commands] Implement "thread siginfo") to 14.x

### ISSUE THREAD

**Issue by mgorny** [llvm/llvm-project#53776]
All patches from the siginfo series except for 287ce6b51675aee43870bebfff68bb144d1ab90e made it into 14.x. Let's backport that one too to complete the feature:

```
commit 287ce6b51675aee43870bebfff68bb144d1ab90e
Author:     Michał Górny <mgorny@moritz.systems>
AuthorDate: 2022-01-28 14:53:20 +0100
Commit:     Michał Górny <mgorny@moritz.systems>
CommitDate: 2022-02-02 19:31:44 +0100

    [lldb] [Commands] Implement "thread siginfo"
    
    Differential Revision: https://reviews.llvm.org/D118473
```

CC @labath, @jimingham

**Comment by labath** [comment]
It's technically not a bug fix, but it's very low-risk, so if we're still at a point where we can pick up changes like this, then I'd definitely recommend it.

**Comment by tstellar** [comment]
Merged: 89fb25f481a5
